Remove `internal_key` from queries filters

GGP1 opened the pull request:
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/30620 |


## Description

Removes fields inside `extra_fields` from queries filters and sorts to avoid exposing internal fields that users cannot retrieve. 

## Tests


<details><summary>Try to filter by internal_key</summary>

```console
wazuh@wazuh:~/wazuh-tools/framework/environments/dev$ curl -H "Authorization: Bearer $TOKEN" -ks https://0.0.0.0:55050/agents?q=internal_key=test | jq
{
  "title": "Bad Request",
  "detail": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key",
  "dapi_errors": {
    "master-node": {
      "error": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key"
    }
  },
  "error": 1408
}
```

</details>

<details><summary>Try to sort by internal_key</summary>

```console
wazuh@wazuh:~/wazuh-tools/framework/environments/dev$ curl -H "Authorization: Bearer $TOKEN" -ks https://0.0.0.0:55050/agents?sort=internal_key | jq
{
  "title": "Bad Request",
  "detail": "Not a valid sort field : Allowed sort fields: ['configSum', 'dateAdd', 'disconnection_time', 'group', 'group_config_status', 'id', 'ip', 'lastKeepAlive', 'manager', 'mergedSum', 'name', 'node_name', 'os.arch', 'os.build', 'os.codename', 'os.major', 'os.minor', 'os.name', 'os.platform', 'os.uname', 'os.version', 'registerIP', 'status', 'status_code', 'version']. Fields: internal_key",
  "remediation": "Please, use only allowed sort fields",
  "dapi_errors": {
    "master-node": {
      "error": "Not a valid sort field : Allowed sort fields: ['configSum', 'dateAdd', 'disconnection_time', 'group', 'group_config_status', 'id', 'ip', 'lastKeepAlive', 'manager', 'mergedSum', 'name', 'node_name', 'os.arch', 'os.build', 'os.codename', 'os.major', 'os.minor', 'os.name', 'os.platform', 'os.uname', 'os.version', 'registerIP', 'status', 'status_code', 'version']. Fields: internal_key"
    }
  },
  "error": 1403
}
```

</details>

<details><summary>Try to select the internal key field</summary>

```console
wazuh@wazuh:~/wazuh-tools/framework/environments/dev$ curl -H "Authorization: Bearer $TOKEN" -ks https://0.0.0.0:55050/agents?select=internal_key | jq
{
  "title": "Bad Request",
  "detail": "Not a valid select field: Allowed select fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Fields internal_key",
  "remediation": "Please, use only allowed select fields",
  "dapi_errors": {
    "master-node": {
      "error": "Not a valid select field: Allowed select fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Fields internal_key"
    }
  },
  "error": 1724
}
```

</details>

<details><summary>test_agent_GET_endpoints.tavern.yaml</summary>

```console
(venv) wazuh@wazuh:~/wazuh/api/test/integration$ pytest --nobuild --disable-warnings test_agent_GET_endpoints.tavern.yaml
============================================ test session starts =============================================
platform linux -- Python 3.10.0, pytest-7.3.1, pluggy-1.6.0
rootdir: /home/wazuh/wazuh/api/test/integration
configfile: pytest.ini
plugins: html-2.1.1, cov-4.1.0, metadata-3.1.1, anyio-4.1.0, trio-0.8.0, asyncio-0.18.1, tavern-1.23.5
asyncio: mode=auto
collected 96 items

test_agent_GET_endpoints.tavern.yaml ................................................................. [ 67%]
...............................                                                                        [100%]

================================ 96 passed, 97 warnings in 293.31s (0:04:53) =================================
```

</details>

fdalmaup reviewed (changes requested):
The `PUT /agents/upgrade` endpoint also works as expected, the problem is that the `internal_key` field appears as an `Available field`, which might be misleading.
```
curl -H "Authorization: Bearer $TOKEN" -ki -X PUT "https://0.0.0.0:55000/agents/upgrade?agents_list=001&q=internal_key=34f016324b9cdb7f6011f8647e5cd2c52caa54e2cf9f2a73f35e5774b1643958"     
HTTP/1.1 400 Bad Request
date: Wed, 02 Jul 2025 19:39:33 GMT
content-type: application/problem+json; charset=utf-8
content-length: 780

{
    "title": "Bad Request",
    "detail": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key",
    "dapi_errors": {
        "wazuh-worker1": {
            "error": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key"
        }
    },
    "error": 1408
}
```

GGP1 commented:
> The `PUT /agents/upgrade` endpoint also works as expected, the problem is that the `internal_key` field appears as an `Available field`, which might be misleading.
> 
> ```
> curl -H "Authorization: Bearer $TOKEN" -ki -X PUT "https://0.0.0.0:55000/agents/upgrade?agents_list=001&q=internal_key=34f016324b9cdb7f6011f8647e5cd2c52caa54e2cf9f2a73f35e5774b1643958"     
> HTTP/1.1 400 Bad Request
> date: Wed, 02 Jul 2025 19:39:33 GMT
> content-type: application/problem+json; charset=utf-8
> content-length: 780
> 
> {
>     "title": "Bad Request",
>     "detail": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key",
>     "dapi_errors": {
>         "wazuh-worker1": {
>             "error": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key"
>         }
>     },
>     "error": 1408
> }
> ```

Good catch, fixed it in https://github.com/wazuh/wazuh/pull/30637/commits/9ce503f0cc3ff8ec1cc3bfd9c4ba2562245bad50 and introduced some extra small enhancements:

- Removed the `internal_key` field from the select error message as well
- Removed an empty space after the sort error message (`Not a valid sort field :` -> `Not a valid sort field:`)
- Converted the fields in the sort error message to a string instead of a list, to be consistent with the other messages
- Ordered the fields alphabetically to show them always in the same order (in sort it was random)
- Fixed a typo in the select error message (`Fields internal_key` -> `Fields: internal_key`)

GGP1 commented:
> The `PUT /agents/upgrade` endpoint also works as expected, the problem is that the `internal_key` field appears as an `Available field`, which might be misleading.
> 
> ```
> curl -H "Authorization: Bearer $TOKEN" -ki -X PUT "https://0.0.0.0:55000/agents/upgrade?agents_list=001&q=internal_key=34f016324b9cdb7f6011f8647e5cd2c52caa54e2cf9f2a73f35e5774b1643958"     
> HTTP/1.1 400 Bad Request
> date: Wed, 02 Jul 2025 19:39:33 GMT
> content-type: application/problem+json; charset=utf-8
> content-length: 780
> 
> {
>     "title": "Bad Request",
>     "detail": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key",
>     "dapi_errors": {
>         "wazuh-worker1": {
>             "error": "Field does not exist: Available fields: id, name, ip, status, os.name, os.version, os.platform, version, manager, dateAdd, group, mergedSum, configSum, os.codename, os.major, os.minor, os.uname, os.arch, os.build, node_name, lastKeepAlive, internal_key, registerIP, disconnection_time, group_config_status, status_code. Field: internal_key"
>         }
>     },
>     "error": 1408
> }
> ```

Good catch, fixed it in https://github.com/wazuh/wazuh/pull/30637/commits/9ce503f0cc3ff8ec1cc3bfd9c4ba2562245bad50 and introduced some extra small enhancements:

- Removed the `internal_key` field from the select error message as well
- Removed an empty space after the sort error message (`Not a valid sort field :` -> `Not a valid sort field:`)
- Converted the fields in the sort error message to a string instead of a list, to be consistent with the other messages
- Ordered the fields alphabetically to show them always in the same order
- Fixed a typo in the select error message (`Fields internal_key` -> `Fields: internal_key`)

fdalmaup reviewed (approved):
LGTM!
```
curl -H "Authorization: Bearer $TOKEN" -ki -s --fail -X PUT 'https://0.0.0.0:55000/agents/upgrade?agents_list=001&q=internal_key=<INTERNAL_KEY>'
{
  "title": "Bad Request",
  "detail": "Field does not exist: Available fields: configSum, dateAdd, disconnection_time, group, group_config_status, id, ip, lastKeepAlive, manager, mergedSum, name, node_name, os.arch, os.build, os.codename, os.major, os.minor, os.name, os.platform, os.uname, os.version, registerIP, status, status_code, version. Field: internal_key",
  "dapi_errors": {
    "wazuh-worker1": {
      "error": "Field does not exist: Available fields: configSum, dateAdd, disconnection_time, group, group_config_status, id, ip, lastKeepAlive, manager, mergedSum, name, node_name, os.arch, os.build, os.codename, os.major, os.minor, os.name, os.platform, os.uname, os.version, registerIP, status, status_code, version. Field: internal_key"
    }
  },
  "error": 1408
}
```
